check_api/src/main/java/com/google/errorprone/dataflow/nullnesspropagation/NullnessAnnotations.java

```diff
@@ -216,7 +222,10 @@ public static Optional<Nullness> getUpperBound(TypeVariable typeVar) {
 
   private static Optional<Nullness> fromAnnotationStream(
       Stream<? extends AnnotationMirror> annotations) {
-    return fromAnnotationSimpleNames(annotations.map(a -> simpleName(a).toString()));
+    return fromAnnotationSimpleNames(
+        annotations
+            .filter(NullnessAnnotations::isAnnotationRelevant)
```

Comment:
Alternatively, perhaps `simpleName` could be taught to return `Optional.empty()` in equivalent cases. A bigger alteration to existing code but at least nominally prevents calling `simpleName` before establishing the precondition.

Reply:
Oh, that doesn't work so well with `annotationsRelevantToNullness` which returns the `AnnotationMirror` (`AnnotationTree`) rather than the simple name&hellip;